### PR TITLE
CORRECTION: Rename CMake package to ROCmCMakeBuildTools

### PR DESCRIPTION
**Pull request by lawruble13**
I incorrectly merged this earlier today, and have undone it. Please see #128 for earlier discussion.

@cgmb @pfultz2 There's been some desire to get this in for ROCm 6.0 regardless, is the state of this okay to be merged from your perspective?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,18 +9,21 @@ set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "")
...
+)
 
 set(CPACK_RPM_PACKAGE_LICENSE "MIT")
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/share/rocm/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/share/rocmcmakebuildtools/cmake)
 include(ROCMCreatePackage)
 include(ROCMSetupVersion)
 
 rocm_setup_version(VERSION 0.10.0)
```
Comment (pfultz2) [inline CMakeLists.txt:23]: Should the version be updated here?

**Comment by pfultz2** [comment]
Shouldn't this go into develop first and then cherrypicked into the release branch?

**Review by pfultz2 — APPROVED** [review]

**Review by cgmb — APPROVED** [review]
Great. We'll formally deprecate the old name in a future release.

**Comment by pramenku** [comment]
Instead of directly changing the name, better to enable deprecation message first and later, we can remove after 1 release.
Due to this change, hipfort and Migraphx compilation failed.

http://rocm-ci.amd.com/blue/rest/organizations/jenkins/pipelines/rocm-dkms-component-staging-parallel-build-stage2libs/runs/26247/nodes/113/log/?start=0

http://rocm-ci.amd.com/blue/rest/organizations/jenkins/pipelines/rocm-dkms-component-staging-parallel-build-stage2libs/runs/26247/nodes/115/log/?start=0

2023-10-12T08:53:49.919Z] -- Detecting CXX compile features - done
[2023-10-12T08:53:49.919Z] CMake Warning (dev) at CMakeLists.txt:17 (find_package):
[2023-10-12T08:53:49.919Z]   Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
[2023-10-12T08:53:49.919Z]   Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
[2023-10-12T08:53:49.919Z]   command to set the policy and suppress this warning.
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z]   Environment variable ROCM_ROOT is set to:
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z]     /opt/rocm-6.0.0-1305
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z]   For compatibility, CMake is ignoring the variable.
[2023-10-12T08:53:49.919Z] This warning is for project developers.  Use -Wno-dev to suppress it.
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] -- Could NOT find ROCmCMakeBuildTools (missing: ROCmCMakeBuildTools_DIR)
[2023-10-12T08:53:49.919Z] CMake Error at CMakeLists.txt:19 (include):
[2023-10-12T08:53:49.919Z]   include could not find requested file:
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z]     ROCMSetupVersion
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] CMake Error at CMakeLists.txt:20 (include):
[2023-10-12T08:53:49.919Z]   include could not find requested file:
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z]     ROCMCreatePackage
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] CMake Error at CMakeLists.txt:25 (rocm_setup_version):
[2023-10-12T08:53:49.919Z]   Unknown CMake command "rocm_setup_version".
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] 
[2023-10-12T08:53:49.919Z] -- Configuring incomplete, errors occurred!

**Comment by cgmb** [comment]
> nstead of directly changing the name, better to enable deprecation message first and later, we can remove after 1 release.
> Due to this change, hipfort and Migraphx compilation failed.

The change was supposed to be purely additive. I've backed the cherrypick out of the ROCm 6.0 staging branch as it seems that this change needs more time for testing.

**Pull request merged**